internal/clients/team4/speaker.go

```diff
@@ -11,21 +11,67 @@ type speaker struct {
 	parent *client
 }
 
-//s.parent.ServerReadHandle.GetGameConfig()
+//SpeakerActionPriorities indicate speaker actions in order of priority
+var SpeakerActionPriorities = []string{
+	"SetVotingResult",
+	"SetRuleToVote",
+	"AnnounceVotingResult",
+	"UpdateRules",
+	"AppointNextJudge",
+}
+
+func (s *speaker) GetSpeakerBudget() shared.Resources {
+	return s.parent.ServerReadHandle.GetGameState().IIGORolesBudget[shared.Speaker]
+}
+
+func (s *speaker) GetActionsCost(actions []string) shared.Resources {
+	gameconfig := s.parent.ServerReadHandle.GetGameConfig().IIGOClientConfig
+	costs := map[string]shared.Resources{
+		"SetVotingResult":      gameconfig.SetVotingResultActionCost,
+		"SetRuleToVote":        gameconfig.SetRuleToVoteActionCost,
+		"AnnounceVotingResult": gameconfig.AnnounceVotingResultActionCost,
+		"UpdateRules":          gameconfig.UpdateRulesActionCost,
+		"AppointNextJudge":     gameconfig.AppointNextJudgeActionCost,
+	}
+	var SumOfCosts shared.Resources = 0
+	for _, action := range actions {
+		SumOfCosts += costs[action]
+	}
+	return SumOfCosts
+}
+
+func (s *speaker) GetHigherPriorityActionsCost(baseaction string) shared.Resources {
+	actionindex := len(SpeakerActionPriorities)
```

Comment:
Initialise to 0. If action is never equal to baseaction, the return at the end will create an out of bounds error

Reply:
getActionsCost already initialises to 0 if it gets an empty array. There's no reason to initialise to 0 twice.

Reply:
Sorry, I see now that it works. I thought [:actionIndex] would through an error if actionIndex was larger than the list, but I see now that that slice is noninclusive the last value.